src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_workload_commands.py

```diff
@@ -0,0 +1,916 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+
+from azure.cli.testsdk import ScenarioTest
+import json
+
+
+id_sql = '/subscriptions/da364f0f-307b-41c9-9d47-b7413ec45535/resourceGroups/pstestwlRG1bca8/providers/Microsoft.Compute/virtualMachines/pstestwlvm1bca8'
+id_hana = '/subscriptions/e3d2d341-4ddb-4c5d-9121-69b7e719485e/resourceGroups/IDCDemo/providers/Microsoft.Compute/virtualMachines/HANADemoIDC3'
+item_id_sql = '/Subscriptions/da364f0f-307b-41c9-9d47-b7413ec45535/resourceGroups/pstestwlRG1bca8/providers/Microsoft.RecoveryServices/vaults/pstestwlRSV1bca8/backupFabrics/Azure/protectionContainers/vmappcontainer;compute;pstestwlrg1bca8;pstestwlvm1bca8/protectedItems/sqldatabase;mssqlserver;testdb1'
+item_id_hana = '/Subscriptions/e3d2d341-4ddb-4c5d-9121-69b7e719485e/resourceGroups/IDCDemo/providers/Microsoft.RecoveryServices/vaults/IDCDemoVault/backupFabrics/Azure/protectionContainers/vmappcontainer;compute;IDCDemo;HANADemoIDC3/protectedItems/SAPHanaDatabase;h22;h22'
+sub_sql = 'da364f0f-307b-41c9-9d47-b7413ec45535'
+sub_hana = 'e3d2d341-4ddb-4c5d-9121-69b7e719485e'
+rg_sql = 'pstestwlRG1bca8'
+rg_hana = 'IDCDemo'
+vault_sql = 'pstestwlRSV1bca8'
+vault_hana = 'IDCDemoVault'
+container_sql = 'VMAppContainer;Compute;pstestwlRG1bca8;pstestwlvm1bca8'
+container_hana = 'VMAppContainer;Compute;IDCDemo;HANADemoIDC3'
+container_friendly_sql = 'pstestwlvm1bca8'
+container_friendly_hana = 'HANADemoIDC3'
+item_auto_sql = 'SQLInstance;mssqlserver'
+item_auto_hana = 'SAPHanaSystem;H22'
+item1_sql = 'SQLDataBase;MSSQLSERVER;testdb1'
+item2_sql = 'msdb'
+item1_hana = 'SAPHanaDatabase;H22;h22'
+item2_hana = 'SYSTEMDB'
+backup_entity_friendly_name_hana = 'H22/H22 [HANADemoIDC3]'
+backup_entity_friendly_name_sql = 'MSSQLSERVER/testdb1 [pstestwlvm1bca8]'
+item_id_hana_2 = '/Subscriptions/e3d2d341-4ddb-4c5d-9121-69b7e719485e/resourceGroups/IDCDemo/providers/Microsoft.RecoveryServices/vaults/IDCDemoVault/backupFabrics/Azure/protectionContainers/vmappcontainer;compute;IDCDemo;HANADemoIDC3/protectedItems/SAPHanaDatabase;h22;h22_restored_sarath'
+
+
+class BackupTests(ScenarioTest, unittest.TestCase):
```

Comment:
If your test are expensive or time consuming, please make it record_only. One more thing to confirm is the resources created by Recovery Service in your test will enable soft delete property by default, right?

Reply:
I'll make it record only. And to answer your second question, the added tests don't create any more resources. We use existing resource to test it. So it might not be possible to test them in live  mode. There are no CLI commands to create a SAPHana server, Hence testing was done this way.